File: fsharp/euler002.fs
module euler002

let answer = 4613732

let result =
    Seq.unfold (fun (c,n) -> Some(c, (n, c + n))) (0,1)
    |> Seq.takeWhile(fun x -> x <= 4000000)
    |> Seq.filter (fun x -> x % 2 = 0)
    |> Seq.sum

File: erlang/euler002.erl
-module(euler002).
-export([result/0]).

fib(0) -> 1;
fib(1) -> 1;
fib(X) -> fib(X-1) + fib(X-2).

result() -> lists:foldl(fun (X, Sum) -> X + Sum end, 0, [X || X <- lists:map(fun (X) -> fib(X) end, lists:seq(0,40)), (X rem 2 == 0) and (X < 4000000)]).